src/Parser/Warlock/Affliction/Modules/Features/AlwaysBeCasting.js

```diff
@@ -6,57 +6,28 @@ import SPELLS from 'common/SPELLS';
 import { formatPercentage } from 'common/format';
 import { STATISTIC_ORDER } from 'Main/StatisticBox';
 import SpellLink from 'common/SpellLink';
+import Wrapper from 'common/Wrapper';
 
 class AlwaysBeCasting extends CoreAlwaysBeCasting {
-  static ABILITIES_ON_GCD = [
-    SPELLS.AGONY.id,
-    SPELLS.CORRUPTION_CAST.id,
-    SPELLS.DEMONIC_GATEWAY_CAST.id,
-    SPELLS.DRAIN_SOUL.id,
-    SPELLS.HEALTH_FUNNEL_CAST.id,
-    SPELLS.LIFE_TAP.id,
-    SPELLS.REAP_SOULS.id,
-    SPELLS.SEED_OF_CORRUPTION_DEBUFF.id,
-    SPELLS.SOULSTONE.id,
-    SPELLS.UNENDING_RESOLVE.id,
-    SPELLS.UNSTABLE_AFFLICTION_CAST.id,
-    // talents
-    SPELLS.HAUNT_TALENT.id,
-    SPELLS.DEMONIC_CIRCLE_TALENT_SUMMON.id,
-    SPELLS.DEMONIC_CIRCLE_TALENT_TELEPORT.id,
-    SPELLS.MORTAL_COIL_TALENT.id,
-    SPELLS.PHANTOM_SINGULARITY_TALENT.id,
-    SPELLS.SOUL_HARVEST_TALENT.id,
-    SPELLS.BURNING_RUSH_TALENT.id,
-    SPELLS.SIPHON_LIFE_TALENT.id,
-    // practically unused, for the sake of completeness
-    SPELLS.SUMMON_DOOMGUARD_TALENTED.id,
-    SPELLS.SUMMON_INFERNAL_TALENTED.id,
-    SPELLS.DARK_PACT_TALENT.id,
-    SPELLS.SUMMON_DOOMGUARD_UNTALENTED.id,
-    SPELLS.SUMMON_INFERNAL_UNTALENTED.id,
-    SPELLS.GRIMOIRE_OF_SACRIFICE_BUFF.id,
-    SPELLS.HOWL_OF_TERROR_TALENT.id,
-    SPELLS.GRIMOIRE_IMP.id,
-    SPELLS.GRIMOIRE_VOIDWALKER.id,
-    SPELLS.GRIMOIRE_FELHUNTER.id,
-    SPELLS.GRIMOIRE_SUCCUBUS.id,
-    SPELLS.SUMMON_IMP.id,
-    SPELLS.SUMMON_VOIDWALKER.id,
-    SPELLS.SUMMON_FELHUNTER.id,
-    SPELLS.SUMMON_SUCCUBUS.id,
-  ];
+  get suggestionThresholds() {
+    return {
+      actual: this.totalTimeWasted / this.owner.fightDuration,
```

Comment:
Please use `this.downtimePercentage` instead. `totalTimeWasted` will be removed soon.